### PR TITLE
failover

### PR DESCRIPTION
**Pull request by shangdehao1**
failure points

### CODE REVIEW

**Comment by shangdehao1** [comment]
"assert" have been modified into "error code", and add some new failure points.

File: src/BackendStore.cpp [hunk 1]
```diff
@@ -171,10 +172,14 @@ int BackendStore::_aio_write( rbd_data* rbd, uint64_t offset, uint64_t length, c
        C_AioBackendCompletion* onfinish ){
     rbd_aio_unit *io_u = new rbd_aio_unit( onfinish );
     int r = rbd_aio_create_completion(io_u, rbc_backend_finish_aiocb, &io_u->completion);
+    if(r<0){
```
Comment (xuechendi) [inline src/BackendStore.cpp]: if (r < 0) better go like this

**Comment by xuechendi** [comment]
LGTM :)

**Comment by zhouyuan** [comment]
need to clean up the commits and make only two:
- adding error status code
- clean up failures on data path

**Comment by zhouyuan** [comment]
squashed into two commits
please pull from https://github.com/zhouyuan/HDCS/tree/rebase_2

**Pull request closed without merging**